sygnal.yaml.sample

```diff
@@ -123,6 +123,31 @@ http:
   #
   port: 5000
 
+## Proxying for outgoing connections #
+#
+proxy:
+  # Specify whether or not to use a proxy for outbound connections.
+  # Currently only HTTP proxies with CONNECT capability are supported.
+  # Specify:
+  # - `true` to use the settings here
+  # - `env` to automatically detect the settings from the de facto standard
+  #   `HTTPS_PROXY`/`https_proxy` environment variables.
+  # - `false` to be strict about not using one, not even from the environment.
+  #
+  # Default is `env`.
+  #
+  enabled: env
```

Comment:
Sure, I suppose I did it because it feels wrong to me to have magic configuration that I can't keep track of, but I suppose anyone with a proxy will prefer to have the proxy declared in one place and everything else 'just work'™.

However, I think it'd be sensible to issue a warning if both are set (or maybe both are set differently?)? Does this sound good?